scio-smb/src/main/scala/com/spotify/scio/smb/util/SMBMultiJoin.scala

```diff
@@ -0,0 +1,3354 @@
+/*
+ * Copyright 2021 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// generated with smb-multijoin.py
+
+package com.spotify.scio.smb.util
+
+import com.spotify.scio.ScioContext
+import com.spotify.scio.coders.Coder
+import com.spotify.scio.values._
+import org.apache.beam.sdk.extensions.smb.{SortedBucketIO, TargetParallelism}
+
+import scala.jdk.CollectionConverters._
+
+final class SMBMultiJoin(@transient private val self: ScioContext) extends Serializable {
```

Comment:
There's no need for this class to be Serializable; we can also remove the `@transient` and make it extend `AnyVal`

Reply:
Extending AnyVal is giving the following compile time error:
` implementation restriction: nested class is not allowed in value class
[error] This restriction is planned to be removed in subsequent releases.
[error]       .map { kv =>`
This would be line 56.
I am still debugging this but please let me know if you have any suggestions for this.

Reply:
Ahh this is in fact an issue with scala 2.12; let's not make it a value class for now; I need to dig if there's a simple workaround.